### PR TITLE
Move from Travis to GitHub Actions

### PR DESCRIPTION
**Pull request by petertseng**
The Exercism organisation is moving to GitHub Actions.
It runs faster.

Closes https://github.com/exercism/d/issues/111

### CODE REVIEW

**Comment by petertseng** [comment]
Proof:

* Fails to compile: https://github.com/petertseng/exercism-d/actions/runs/375857744
* Compiles, but gives wrong output on tests: https://github.com/petertseng/exercism-d/actions/runs/375863164
* Succeeds: https://github.com/petertseng/exercism-d/actions/runs/375864249

**Review by ErikSchierboom — APPROVED** [review]
LGTM! Thanks

**Comment by petertseng** [comment]
This PR has just now been updated to use `main` instead of `master`, which makes it up to date with the repo as of today. The maintainer will notice that CI fails. That is because this PR depends on #149 now. I do not allow myself to include the commit from #149 in this PR because I will not willingly participate in allowing PRs/commits that do too many things. I will however for the maintainer's convenience provide #150 to demonstrate that CI will pass after #149 is merged. The maintainer may request that I rebase this PR after #149 is merged, which I will happily do, or if the maintainer is comfortable doing so, the maintainer has permissions to do so.

File: .github/workflows/tests.yml
Comment (ErikSchierboom) [inline .github/workflows/tests.yml:21]: You could, if you'd like, remove this bit as there is already a [separate workflow](https://github.com/exercism/d/blob/main/.github/workflows/configlet.yml) that does this.

**Review by ErikSchierboom — APPROVED** [review]
Thanks a lot! It will be so useful to get CI passing.

**Comment by petertseng** [comment]
Ah, I've just looked at the teams and there are no members of the D track maintainers team. Well, this PR will just have to wait until one comes along, then. That is fine; it doesn't harm me to leave PRs in my outgoing PRs queue. It will not be possible for me to volunteer to take on this role, due to the limits on my time.

**Comment by ErikSchierboom** [comment]
As there don't seem to be any active maintainers, I'll assume that role and merge this. Thanks @petertseng!

**Pull request merged**
